VideoWeb/VideoWeb.Common/Caching/IConsultationInvitationCache.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VideoWeb.Common.Models;
+
+namespace VideoWeb.Common.Caching
+{
+    public interface IConsultationInvitationCache
+    {
+        Task CreateInvitationEntry(ConsultationInvitation consultationInvitation);
+        Task<ConsultationInvitation> GetInvitation(Guid invitationId);
+        Task UpdateResponseToInvitation(Guid invitationId, Guid participantId, ConsultationAnswer answer);
+        Task DeleteInvitationEntry(Guid invitationId);
```

Comment:
I had a look into this and it introduced another race condition bug. I think it would be possible to do this if the logic in consultations controller was modified.

Reply:
What would you suggest to change in the consultations controller?